packages/logger-plugin/src/action-logger.ts

```diff
@@ -0,0 +1,45 @@
+import { getActionTypeFromInstance, Store } from '@ngxs/store';
+
+import { formatTime } from './internals';
+import { LogWriter } from './log-writer';
+
+export class ActionLogger {
+  constructor(private action: any, private store: Store, private logWriter: LogWriter) {}
+
+  dispatched(state: any) {
+    const actionName = getActionTypeFromInstance(this.action);
+    const formattedTime = formatTime(new Date());
```

Comment:
Why you used `formatTime(new Date())` ? instead

```ts
import { DatePipe } from '@angular/common';

const formattedTime = new DatePipe('en-EU').transform(new Date(), 'HH:mm:ss.SSS');
```

Reply:
Good idea. It was a refactor, so I just extracted code that was already there.

Reply:
I suspect that the current method may be faster than this because it is less generic, and it might be a smaller bundle size too (doesn't bring in the Angular pipe dependency).

Reply:
@angular/common - every where usage in angular app as dependency
also you can check bundle size after refactor (npm run bundlesize)

Reply:
`I suspect that the current method may be faster` 

firstly, the previous code is completely incomprehensible, I had to run to understand what it was doing

secondly, people should not use the plugin in production

```ts
@NgModule({
 imports: [
   environment.production ? [] : [ NgxsLoggerPlugin.forRoot() ]
 ]
})
export class AppModule {}
```

Reply:
I am happy to use the pipe, but please bear in mind that this is a refactor, so I am just changing the structure of the code, not rewriting it.

Reply:
up to you

Reply:
Left it as is.